readium/streamer/src/main/java/org/readium/r2/streamer/parser/image/ImageParser.kt

```diff
@@ -26,18 +24,31 @@ import org.readium.r2.streamer.parser.PublicationParser
  */
 class ImageParser : PublicationParser {
 
-    override suspend fun parse(asset: PublicationAsset, warnings: WarningLogger?): Publication.Builder? {
+    override suspend fun parse(
+        asset: PublicationParser.Asset,
+        warnings: WarningLogger?
+    ): Try<Publication.Builder, PublicationParser.Error> {
 
-        if (!accepts(asset.mediaType, asset.fetcher))
-            return null
+        if (!asset.mediaType.matches(MediaType.CBZ) && !asset.mediaType.isBitmap)
+            return Try.failure(PublicationParser.Error.FormatNotSupported())
 
-        val readingOrder = asset.fetcher.links()
-            .filter { !File(it.href).isHiddenOrThumbs && it.mediaType.isBitmap }
-            .sortedBy(Link::href)
-            .toMutableList()
+        val readingOrder =
+            if (asset.mediaType.matches(MediaType.CBZ)) {
```

Comment:
Same comment as for the ZABs.

Reply:
Here filtering might be critical if the `Fetcher` is not aware of the single file media type.